test/OpenTelemetry.Contrib.Instrumentation.MassTransit.Tests/MassTransitInstrumentationTests.cs

```diff
@@ -55,33 +53,173 @@ public async Task MassTransitInstrumentationConsumerAndHandlerTest()
+                Assert.Equal(expectedMessageContext.DestinationAddress.Host, actualActivity.GetTagValue(SemanticConventions.AttributeNetPeerName)?.ToString());
+
+                Assert.Null(actualActivity.GetTagValue(TagName.MessageId));
+                Assert.Null(actualActivity.GetTagValue(TagName.ConversationId));
+                Assert.Null(actualActivity.GetTagValue(TagName.DestinationAddress));
+
+                Assert.Null(actualActivity.GetTagValue(TagName.SpanKind));
+                Assert.Null(actualActivity.GetTagValue(TagName.PeerService));
+
+                Assert.Null(actualActivity.GetTagValue(TagName.PeerAddress));
+                Assert.Null(actualActivity.GetTagValue(TagName.PeerHost));
+                Assert.Null(actualActivity.GetTagValue(TagName.SourceAddress));
             }
+        }
 
-            foreach (var activity in receives)
+        [Fact]
+        public async Task ShouldMapMassTransitTagsForReceiveMessageToOpenTelemetrySpecification()
+        {
+            var activityProcessor = new Mock<BaseProcessor<Activity>>();
+            using (Sdk.CreateTracerProviderBuilder()
+                .AddProcessor(activityProcessor.Object)
+                .AddMassTransitInstrumentation()
+                .Build())
             {
-                Assert.Equal("RECV /input_queue", activity.DisplayName);
+                var harness = new InMemoryTestHarness();
+                var consumerHarness = harness.Consumer<TestConsumer>();
+                var handlerHarness = harness.Handler<TestMessage>();
+                await harness.Start();
+                try
+                {
+                    await harness.InputQueueSendEndpoint.Send<TestMessage>(new { Text = "Hello, world!" });
+
+                    Assert.True(await harness.Consumed.SelectAsync<TestMessage>().Any());
+                    Assert.True(await consumerHarness.Consumed.SelectAsync<TestMessage>().Any());
+                    Assert.True(await handlerHarness.Consumed.SelectAsync().Any());
+                }
+                finally
+                {
+                    await harness.Stop();
+                }
+
+                var expectedMessageContext = harness.Sent.Select<TestMessage>().FirstOrDefault()?.Context;
+                var actualActivity = this.GetActivitiesFromInvocationsByOperationName(activityProcessor.Invocations, OperationName.Transport.Receive).LastOrDefault();
+
+                Assert.NotNull(actualActivity);
+                Assert.NotNull(expectedMessageContext);
+
+                Assert.Equal("/input_queue consume", actualActivity.DisplayName);
+                Assert.Equal(ActivityKind.Consumer, actualActivity.Kind);
+                Assert.Equal("loopback", actualActivity.GetTagValue(SemanticConventions.AttributeMessagingSystem)?.ToString());
+
+                Assert.Equal(expectedMessageContext.MessageId.ToString(), actualActivity.GetTagValue(SemanticConventions.AttributeMessagingMessageId)?.ToString());
```

Comment:
We could detect, that a destination is `temporary` based on `input-address` tag, for example
 `rabbitmq://localhost/TempQueuName_bus_yyyoyyyxieyyud4xbdc8w34sy1?temporary=true`
--